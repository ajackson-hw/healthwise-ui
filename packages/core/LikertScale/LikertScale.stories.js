import React from 'react'
import { storiesOf } from '@storybook/react'
import LikertScale from './index'

storiesOf('Components/Likert Scale', module)
  .add(
    'with defaults',
    () => <LikertScale id="1" />,
    {
      info: `
        Demonstates basic rendering with defaults
      `
    }
  )
  .add(
    'with selected answer & custom labels',
    () => <LikertScale id="1" selectedAnswer="6" minValueText="Meh." maxValueText="OH YEAH!" />,
    {
      info: `
        Demonstates rendering with preselected answer & custom labels
      `
    }
  )
  .add(
    'with readonly',
    () => <LikertScale id="1" selectedAnswer="8" readonly />,
    {
      info: `
        Demonstates preselected answer in readonly mode
      `
    }
  )
  .add(
    'with range',
    () => <LikertScale id="1" minValue="1" maxValue="5" selectedAnswer="2" />,
    {
      info: `
        Demonstates custom range and preselected answer
      `
    }
  )
  .add(
    'with custom colors',
    () => (
      <LikertScale
        id="1"
        baseRgbColor="150, 50, 53"
        darkTextRgbColor="100, 1, 1"
        lightTextRgbColor="230, 230, 230"
        selectedAnswer="10"
      />
    ),
    {
      info: `
        Demonstates custom color and preselected answer
      `
    }
  )
