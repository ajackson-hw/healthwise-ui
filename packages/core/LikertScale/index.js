import React from 'react'
import PropTypes from 'prop-types'
import styles from './style.css'
import Radio from '@healthwise/radio'

class LikertScale extends React.Component {
  constructor(props) {
    super(props)

    this.handleChange = this.handleChange.bind(this)
    this.getAxisId = this.getAxisId.bind(this)
    this.axisButtonColors = []
    this.axisBgColors = []
    this.axisTextColors = []
    this.classNameBase = 'hw_likert_axis_button_'

    // set up component state so we can retain the selected value (even if the change doesn't trigger a rerender)
    this.state = { currentValue: this.props.selectedAnswer }

    this.initStyle()
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const currentVal = parseInt(this.props.selectedAnswer, 10)
    const newVal = parseInt(nextProps.selectedAnswer, 10)

    if (currentVal !== newVal) {
      this.setState({ currentValue: newVal })
    }
  }

  initStyle() {
    const props = this.props
    const total = props.maxValue - props.minValue + 1
    const increment = 1 / total

    for (let num = 0; num < total; num++) {
      const alpha = increment * num + increment

      this.axisButtonColors[num] = styles[this.classNameBase + num]
      this.axisBgColors[num] = {
        backgroundColor: 'rgba(' + this.props.baseRgbColor + ', ' + alpha + ')',
      }
      this.axisTextColors[num] = { color: this.getTextColor(alpha) }
    }
  }

  getTextColor(alpha) {
    const rgbaToRgb = function(colors, alpha) {
      return colors.map(function(color) {
        return (1 - alpha) * 255 + alpha * color
      })
    }
    // adapted from http://stackoverflow.com/a/3943023
    const getLuminance = function(colors) {
      colors = colors.map(function(c) {
        c = c / 255
        return c <= 0.03928 ? c / 12.92 : Math.pow((c + 0.055) / 1.055, 2.4)
      })
      return 0.2126 * colors[0] + 0.7152 * colors[1] + 0.0722 * colors[2]
    }

    let bg = this.props.baseRgbColor.split(',')
    let darkText = this.props.darkTextRgbColor.split(',')
    let lightText = this.props.lightTextRgbColor.split(',')
    let toReturn = this.props.darkTextRgbColor

    // compare the bg luminance to the dark and light text and pick the best color based on that
    if (
      bg &&
      darkText &&
      lightText &&
      bg.length === 3 &&
      darkText.length === 3 &&
      lightText.length === 3
    ) {
      const bgColor = rgbaToRgb(bg, alpha)
      const bgLuminance = getLuminance(bgColor)
      const darkLuminance = getLuminance(darkText)
      const lightLuminance = getLuminance(lightText)

      // TODO (enhancement): we could determine if their contrast ratio is high enough, and if it's not, just send black/white text
      toReturn =
        (bgLuminance + 0.05) / (darkLuminance + 0.05) >
        (lightLuminance + 0.05) / (bgLuminance + 0.05)
          ? this.props.darkTextRgbColor
          : this.props.lightTextRgbColor
    }

    return 'rgb(' + toReturn + ')'
  }

  handleChange(e) {
    const el = e.currentTarget

    if (!this.props.readonly) {
      if (el) {
        const selectedValue = parseInt(el.getAttribute('data-hw-value'), 10)

        // record the value locally
        this.setState({ currentValue: selectedValue })

        // run callback with selected value
        this.props.onAnswer(e, selectedValue, this.props.answerId)
      }
    } else {
      e.preventDefault()
      e.stopPropagation()
    }
  }

  getAxisId(num) {
    return this.props.id + '_key_' + num
  }

  render() {
    const selectedAnswer = parseInt(this.state.currentValue, 10)
    let {
      id,
      scaleWrapperId,
      name,
      isRequired,
      minValue,
      maxValue,
      minValueText,
      maxValueText,
      readonly,
      forPrint,
    } = this.props

    name = name || id
    scaleWrapperId = scaleWrapperId || id + '_scale_wrapper'

    let labels = []
    let disabledLabel = readonly ? styles.likert_axis_label_disabled + ' ' : ''
    let printLabel = forPrint ? styles.likert_for_print + ' ' : ''
    let width = 100 / (maxValue - minValue + 1) + '%'
    let widthObj = { width: width }

    for (let i = 0, num = minValue; num <= maxValue; i++, num++) {
      let axisId = this.getAxisId(num)
      let classNames = disabledLabel + printLabel + styles.likert_axis_label
      let radio = null
      let textColorObj = this.axisTextColors[this.axisTextColors.length - 1]
      let bgColorObj = { backgroundColor: 'rgba(' + this.props.baseRgbColor + ', 1)' }

      const defaultRadio = (
        <Radio
          forPrint={forPrint}
          onChange={this.handleChange}
          name={name}
          id={axisId}
          data-hw-width={num}
          data-hw-value={num}
          defaultChecked={selectedAnswer === num}
          value={'hw-likert-button_val_' + num}
          isRequired={isRequired}
        />
      )

      if (!readonly) {
        textColorObj = this.axisTextColors[i]
        bgColorObj = this.axisBgColors[i]

        radio = defaultRadio
      }

      if (forPrint) {
        radio = defaultRadio
      }

      if (!isNaN(selectedAnswer) && num <= selectedAnswer) {
        // color all the squares to the left of and including the selected answer
        classNames += ` ${this.axisButtonColors[i]} ${styles.likert_axis_button_active}`
      } else {
        bgColorObj = {}
        textColorObj = { color: 'rgb(' + this.props.darkTextRgbColor + ')' }
      }

      var blockStyles = Object.assign(bgColorObj, widthObj)

      labels.push(
        <div className={classNames} id={axisId + '_label'} key={axisId} style={blockStyles}>
          <label htmlFor={axisId}>
            <span className={styles.likert_axis_label_num} style={textColorObj}>
              {num}
            </span>
            {radio}
          </label>
        </div>
      )
    }

    const parentClasses =
      'hw-likert-scale ' + styles.likert_scale + (readonly ? ' ' + styles.likert_readonly : '')
    const textValueEquivalent = readonly ? (
      <span className={styles.common_hidden_text}>{`${selectedAnswer || 0} of ${maxValue}`}</span>
    ) : null

    return (
      <div
        className={parentClasses}
        id={id}
        data-hw-likert-min={minValue}
        data-hw-likert-max={maxValue}
      >
        <div className={'hw-likert-labels ' + styles.likert_labels}>
          <span
            className={
              'hw-likert-label hw-likert-label-left ' +
              styles.likert_label +
              ' ' +
              styles.likert_label_left
            }
            aria-hidden="true"
          >
            <span className={styles.likert_label_left_span}>{minValueText}</span>
          </span>
          <span
            className={
              'hw-likert-label hw-likert-label-right ' +
              styles.likert_label +
              ' ' +
              styles.likert_label_right
            }
            aria-hidden="true"
          >
            <span className={styles.likert_label_right_span}>{maxValueText}</span>
          </span>
        </div>
        <div
          id={scaleWrapperId}
          className={'hw-likert-scale-wrapper ' + styles.likert_scale_wrapper}
        >
          <div className={'hw-likert-axis ' + styles.likert_axis} aria-hidden={readonly}>
            {labels}
          </div>
          {textValueEquivalent}
        </div>
      </div>
    )
  }
}

LikertScale.propTypes = {
  id: PropTypes.string.isRequired,
  answerId: PropTypes.string,
  isRequired: PropTypes.bool,
  selectedAnswer: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  maxValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minValueText: PropTypes.string,
  maxValueText: PropTypes.string,
  name: PropTypes.string,
  scaleWrapperId: PropTypes.string,
  baseRgbColor: PropTypes.string,
  darkTextRgbColor: PropTypes.string,
  lightTextRgbColor: PropTypes.string,
  readonly: PropTypes.bool,
  forPrint: PropTypes.bool,
  onAnswer: PropTypes.func,
}

LikertScale.defaultProps = {
  isRequired: false,
  minValue: 0,
  maxValue: 10,
  minValueText: 'Not at all important',
  maxValueText: 'Extremely important',
  baseRgbColor: '1, 122, 203',
  darkTextRgbColor: '54, 57, 66',
  lightTextRgbColor: '255, 255, 255',
  readonly: false,
  forPrint: false,
  onAnswer: function() {},
}

export default LikertScale
