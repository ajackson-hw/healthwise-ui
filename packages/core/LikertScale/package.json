{
  "name": "@healthwise/likert-scale",
  "version": "1.0.6",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - likert-scale assets",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@healthwise/radio": "^2.0.0",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "react": ">=15.4.2"
  }
}
