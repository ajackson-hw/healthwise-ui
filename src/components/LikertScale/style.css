.common_hidden_text {
  position: absolute;
  top: auto;
  left: -10000px;
  display: block;
  overflow: hidden;
  width: 1px;
  height: 1px;
  background: transparent;
}

.likert_scale {
  margin: 10px auto 2em;
}

.likert_labels {
  display: table;
  width: 100%;
  padding-bottom: 10px;
}

.likert_label {
  display: table-cell;
  width: 50%;
  text-align: left;
}

.likert_label_left_span,
.likert_label_right_span {
  display: inline-block;
  width: 30%;
  color: #363942;
  font-weight: 500;
  line-height: 1em;
  font-size: 0.75em;
}

.likert_label_right {
  text-align: right;
}

.likert_scale_wrapper {
  width: 100%;
  box-sizing: border-box;
}

.likert_axis {
  display: block;
  width: 100%;
  box-sizing: border-box;
}

.likert_axis_label {
  display: inline-block;
  box-sizing: border-box;
  width: 9.09092%;
  background: #fff;
  font-size: 1.25em;
  text-align: center;
  border: 2px solid #dde0e6;
  border-right: none;
  margin: 0;
  padding: 0;
  transition: all 0.5s ease-out;
}

.likert_axis_label label {
  display: block;
  box-sizing: border-box;
  width: 100%;
  cursor: pointer;
  margin: 0;
  padding: 6px;
}

.likert_axis_label_disabled {
  cursor: default;
  padding-top: 27px;
  height: 74px;
}

.likert_for_print {
  padding-top: 16px;
  height: 80px;
}

.likert_axis_label_num {
  display: block;
  padding-top: 0.4em;
}

.likert_axis_label_disabled .likert_axis_label_num {
  padding: 0;
}

.likert_axis_label:first-child {
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.likert_axis_label:last-child {
  border-right: 2px solid #dde0e6;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
}

.likert_axis_label input[type="radio"] {
  display: block;
  margin: 15px auto 5px;
  padding: 4px 2px 2px 4px;
}

@media screen and (max-width: 750px) {
  .likert_scale {
    margin-bottom: 1em;
  }

  .likert_axis_label {
    border-width: 1px;
    padding: 0;
  }

  .likert_axis_label_disabled {
    padding-top: 0;
    height: inherit;
    line-height: 2em;
  }

  .likert_label_left_span,
  .likert_label_right_span {
    width: 35%; /* so the labels don't get weird on smaller screens */
  }

  .likert_axis_label:first-child {
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
  }

  .likert_axis_label:last-child {
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px;
  }
}

/* Targets displays using Windowsâ€™ High Contrast Mode themes: */

@media screen and (-ms-high-contrast: active) {
  .likert_axis_button_active {
    background-color: #fff;
    border-color: #fff;
    color: #000;
  }

  .likert_axis_label :global(.hw-radio-wrapper .hw-radio-image .hw-radio-focus-outline) {
    stroke: #fff;
  }

  .likert_axis_button_active :global(.hw-radio-wrapper .hw-radio-image .hw-radio-focus-outline) {
    stroke: #000;
  }

  .likert_readonly .likert_axis_button_active {
    background-color: #fff;
    border-color: #fff;
    color: #000;
  }
}

@media screen and (-ms-high-contrast: black-on-white) {
  .likert_axis_button_active {
    background-color: #000;
    border-color: #000;
    color: #fff;
  }

  .likert_axis_label :global(.hw-radio-wrapper .hw-radio-image .hw-radio-focus-outline) {
    stroke: #000;
  }

  .likert_axis_button_active :global(.hw-radio-wrapper .hw-radio-image .hw-radio-focus-outline) {
    stroke: #fff;
  }

  .likert_readonly .likert_axis_button_active {
    background-color: #000;
    border-color: #000;
    color: #fff;
  }
}

@media print {

  .likert_axis_label_disabled,
  .likert_axis_label_disabled:first-child,
  .likert_axis_label_disabled:last-child {
    padding: 0.25em;
    border-color: #eee;
    height: auto;
    width: 9.09092%;
    font-weight: normal;
    color: #9e9e9e;
  }

  .likert_readonly .likert_axis_button_active {
    font-weight: bold;
    border-color: rgba(1, 122, 203, 1);
    border-right: 2px solid rgba(1, 122, 203, 1);
    border-left-width: 0;
    background-color: #fff;
    color: #000;
  }

  .likert_readonly .hw_likert_axis_button_0 {
    border-left-width: 2px;
  }
}
