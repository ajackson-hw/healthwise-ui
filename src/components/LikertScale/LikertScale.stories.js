import React from 'react'
import { storiesOf } from '@storybook/react'
import LikertScale from './index'

storiesOf('Components/Likert Scale', module)
  .addWithInfo(
    'with defaults',
    `
      Demonstates basic rendering with defaults
    `,
    () => <LikertScale id="1" />
  )
  .addWithInfo(
    'with selected answer & custom labels',
    `
      Demonstates rendering with preselected answer & custom labels
    `,
    () => <LikertScale id="1" selectedAnswer="6" minValueText="Meh." maxValueText="OH YEAH!" />
  )
  .addWithInfo(
    'with readonly',
    `
      Demonstates preselected answer in readonly mode
    `,
    () => <LikertScale id="1" selectedAnswer="8" readonly />
  )
  .addWithInfo(
    'with range',
    `
      Demonstates custom range and preselected answer
    `,
    () => <LikertScale id="1" minValue="1" maxValue="5" selectedAnswer="2" />
  )
  .addWithInfo(
    'with custom colors',
    `
      Demonstates custom color and preselected answer
    `,
    () => (
      <LikertScale
        id="1"
        baseRgbColor="150, 50, 53"
        darkTextRgbColor="100, 1, 1"
        lightTextRgbColor="230, 230, 230"
        selectedAnswer="10"
      />
    )
  )
