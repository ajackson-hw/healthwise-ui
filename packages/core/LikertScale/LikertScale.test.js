/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'
import LikertScale from './index'

// TODO: This has minimal testing because when we go to really use it we may need to strip out the event functionality

describe('<LikertScale />', () => {
  it('renders correctly with minimal attributes', () => {
    const component = renderer.create(<LikertScale id="q1" name="q1_0u" />)
    expect(component).toMatchSnapshot()
  })
})
