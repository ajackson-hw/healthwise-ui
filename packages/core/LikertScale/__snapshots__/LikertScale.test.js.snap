// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikertScale /> renders correctly with minimal attributes 1`] = `
<div
  className="hw-likert-scale likert_scale"
  data-hw-likert-max={10}
  data-hw-likert-min={0}
  id="q1"
>
  <div
    className="hw-likert-labels likert_labels"
  >
    <span
      aria-hidden="true"
      className="hw-likert-label hw-likert-label-left likert_label likert_label_left"
    >
      <span
        className="likert_label_left_span"
      >
        Not at all important
      </span>
    </span>
    <span
      aria-hidden="true"
      className="hw-likert-label hw-likert-label-right likert_label likert_label_right"
    >
      <span
        className="likert_label_right_span"
      >
        Extremely important
      </span>
    </span>
  </div>
  <div
    className="hw-likert-scale-wrapper likert_scale_wrapper"
    id="q1_scale_wrapper"
  >
    <div
      aria-hidden={false}
      className="hw-likert-axis likert_axis"
    >
      <div
        className="likert_axis_label"
        id="q1_key_0_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_0"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            0
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={0}
              data-hw-width={0}
              defaultChecked={false}
              disabled={false}
              id="q1_key_0"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_0"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_1_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_1"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            1
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={1}
              data-hw-width={1}
              defaultChecked={false}
              disabled={false}
              id="q1_key_1"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_1"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_2_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_2"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            2
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={2}
              data-hw-width={2}
              defaultChecked={false}
              disabled={false}
              id="q1_key_2"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_2"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_3_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_3"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            3
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={3}
              data-hw-width={3}
              defaultChecked={false}
              disabled={false}
              id="q1_key_3"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_3"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_4_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_4"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            4
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={4}
              data-hw-width={4}
              defaultChecked={false}
              disabled={false}
              id="q1_key_4"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_4"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_5_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_5"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            5
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={5}
              data-hw-width={5}
              defaultChecked={false}
              disabled={false}
              id="q1_key_5"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_5"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_6_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_6"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            6
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={6}
              data-hw-width={6}
              defaultChecked={false}
              disabled={false}
              id="q1_key_6"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_6"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_7_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_7"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            7
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={7}
              data-hw-width={7}
              defaultChecked={false}
              disabled={false}
              id="q1_key_7"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_7"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_8_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_8"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            8
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={8}
              data-hw-width={8}
              defaultChecked={false}
              disabled={false}
              id="q1_key_8"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_8"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_9_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_9"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            9
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={9}
              data-hw-width={9}
              defaultChecked={false}
              disabled={false}
              id="q1_key_9"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_9"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
      <div
        className="likert_axis_label"
        id="q1_key_10_label"
        style={
          Object {
            "width": "9.090909090909092%",
          }
        }
      >
        <label
          htmlFor="q1_key_10"
        >
          <span
            className="likert_axis_label_num"
            style={
              Object {
                "color": "rgb(54, 57, 66)",
              }
            }
          >
            10
          </span>
          <span
            className="hw-radio-wrapper radio_wrapper"
          >
            <input
              className="hw-radio radio"
              data-hw-value={10}
              data-hw-width={10}
              defaultChecked={false}
              disabled={false}
              id="q1_key_10"
              name="q1_0u"
              onChange={[Function]}
              onClick={undefined}
              required={false}
              style={null}
              type="radio"
              value="hw-likert-button_val_10"
            />
            <svg
              className="hw-radio-image radio_image"
              focusable="false"
              role="presentation"
              style={null}
              viewBox="0 0 28 28"
            >
              <circle
                className="hw-radio-outer-circle radio_outer_circle"
                cx="14"
                cy="14"
                r="9"
                style={null}
              />
              <circle
                className="hw-radio-unchecked radio_inner_circle radio_unchecked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
              <circle
                className="hw-radio-checked radio_inner_circle radio_checked"
                cx="14"
                cy="14"
                r="5"
                style={null}
              />
            </svg>
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
